### PR TITLE
#163518645 Social Auth

### PR DESCRIPTION
**Pull request by abayo-luc**
#### What does this PR do?
Add social auth
#### Description of Task to be completed?
- Activated social login button
- A user can log in with Google, Facebook, or Twitter
#### How should this be manually tested?
- clone the repo and check out the following branch `ft-social-auth-163518645 `
- Also, clone the backed repo and check out branch `ch-server-message-165307938`
- run `yarn start` and navigate to /sign_in in your browser
- Try one of the social auth buttons
### Related Pivotal tracker ID
#163518645

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -50,6 +50,8 @@
     "no-named-as-default": "off",
     "arrow-body-style": ["error", "as-needed"],
     "import/no-dynamic-require": "off",
+    "react/forbid-prop-types": "off",
```
Comment (Musigwa-zz) [inline .eslintrc.json]: @abayo-luc, is there a meaningful reason why you disabled this rule?

File: .eslintrc.json [hunk 2]
```diff
@@ -50,6 +50,8 @@
...
+    "jsx-a11y/click-events-have-key-events": "off",
```
Comment (Musigwa-zz) [inline .eslintrc.json]: @abayo-luc, do you mind checking out this [reason](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/click-events-have-key-events.md) why we should use this rule?

**Review by Musigwa-zz — CHANGES_REQUESTED** [review]
Good work @abayo-luc, please consider the comments I left on some of the files changes.

**Review by Musigwa-zz — APPROVED** [review]

**Review by niomwungeri-fabrice — APPROVED** [review]
Good Job mate!

**Review by puritybirir — APPROVED** [review]

**Pull request merged**